### PR TITLE
Fix bugs in Delete and Edit.

### PR DESCRIPTION
**Pull request by difanyang**
These functions should now work properly.

### CODE REVIEW

File: easylinks/src/main/java/com/google/step/servlets/ManageServlet.java [hunk 1]
```diff
@@ -72,7 +73,35 @@ private static void addLink(HttpServletRequest request, HttpServletResponse resp
...
+    if (shortcut.isEmpty() || shortcut.startsWith(PROVIDED_LINK) || url.isEmpty()) {
+      response.getWriter().println("Invalid Input.");
+      return;
+    }
+
+    deleteLinkWithId(id);
+    Entity easyLinkEntity = new Entity("Link");
+    easyLinkEntity.setProperty("shortcut", shortcut);
+    easyLinkEntity.setProperty("url", url);
+    easyLinkEntity.setProperty("email", ServletHelper.USERSERVICE.getCurrentUser().getEmail());
```
Comment (xiaoluLucy814) [inline easylinks/src/main/java/com/google/step/servlets/ManageServlet.java:91]: Maybe don't need to delete the entity first and then add? If you set the key of easyLinkEntity to be id, I think the old entity can be overwritten.

Comment (difanyang) [reply easylinks/src/main/java/com/google/step/servlets/ManageServlet.java]: Done.

File: easylinks/src/main/java/com/google/step/servlets/ManageServlet.java [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package com.google.step.servlets;
 
 import com.google.step.servlets.ServletHelper;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
```
Comment (xiaoluLucy814) [inline easylinks/src/main/java/com/google/step/servlets/ManageServlet.java:5]: Consider removing unnecessary libraries

Comment (difanyang) [reply easylinks/src/main/java/com/google/step/servlets/ManageServlet.java]: Oh I added that when trying to throw EntityNotFoundException and forgot to remove. Done.

**Pull request merged**